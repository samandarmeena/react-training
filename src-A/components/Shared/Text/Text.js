import React from 'react';
import styles from './Text.module.scss';
import PropTypes from 'prop-types';

/**
 * Text component
 * It renders text
 * @param {string} text
 * @param {boolean} isStrong
 * @return Element
 */
const Text = (props) => {
  const { text, isStrong } = props;
  const classStrong = isStrong ? styles.textStrong : '';
  return <span className={classStrong}>{text}</span>;
};

// proptype validations for props
Text.propTypes = {
  text: PropTypes.string.isRequired,
  isStrong: PropTypes.bool
};

// default values for props
Text.defaultProps = {
  isStrong: false
};

export default Text;
