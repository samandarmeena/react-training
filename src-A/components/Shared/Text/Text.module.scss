$textStrong: 700;
.textStrong {
  font-weight: $textStrong;
}
