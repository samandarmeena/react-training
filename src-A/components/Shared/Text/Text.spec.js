import React from 'react';
import { mount } from 'enzyme';
import Text from './Text';

describe('Text component', () => {
  let wrapper = '';

  beforeAll(() => {
    wrapper = mount(<Text text="Terms and Conditions" isStrong={true} />);
  });

  it('Text component rendered without an error', () => {
    expect(wrapper).toBeTruthy();
  });

  /*it('Check passed props', () => {
    expect(wrapper.prop('text')).toBe('Terms and Conditions');
    expect(wrapper.prop('isStrong')).toBe(true);
  });*/

  it('Check length of the children', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('Text is rendering properly', () => {
    expect(wrapper.find('span').text()).toBe('Terms and Conditions');
  });

  it('Check dynamic class', () => {
    expect(wrapper.find('span').hasClass('textStrong')).toBe(true);
  });

  it('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
