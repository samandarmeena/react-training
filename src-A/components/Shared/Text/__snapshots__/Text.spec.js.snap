// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Snapshot testing 1`] = `
<Text
  isStrong={true}
  text="Terms and Conditions"
>
  <span
    className="textStrong"
  >
    Terms and Conditions
  </span>
</Text>
`;
